facilitator/src/gcp_oauth.rs

```diff
@@ -186,14 +225,15 @@ impl ProvideDefaultToken for ServiceAccountKeyFileDefaultTokenProvider {
 
 /// Uses a GCP Workload Identity Pool to federate GCP IAM with AWS IAM, allowing
 /// workloads in AWS Elastic Kubernetes Service to impersonate a GCP SA.
+#[derive(Clone)]
 struct AwsIamFederationViaWorkloadIdentityPoolDefaultTokenProvider {
-    aws_credentials_provider: Box<dyn ProvideAwsCredentials>,
+    aws_credentials_provider: aws_credentials::Provider,
```

Comment:
I would personally feel more comfortable putting this in an Arc if we're going to clone this struct. I did read that internally they're using Arc, but if they make an implementation mistake in the future this is going to crash on its head.

Although, I have no idea what kind of performance impact making this Arc would have.

Either way, I think it's fine whatever decision you make.

Reply:
That's an interesting idea. Even if we did wrap every variant of `aws_credentials::Provider` (which is a type we define) in `std::sync::Arc`, all that would guarantee is that we can call `clone()` and have... something happen. It wouldn't protect us against a change in the implementation of say `rusoto_core::credentials::DefaultCredentialsProvider` that causes it to no longer use an `Arc<Mutex<>>` internally to efficiently share its state across instances. In fact, wrapping stuff in `Arc` would make it harder for us to detect that Rusoto had made a change such that any of `DefaultCredentialsProvider`, `AutoRefreshingProvider` or `WebIdentityProvider` was no longer `Clone`.

I'd argue that generally speaking, we can't always effectively defend ourselves today against arbitrary performance or correctness regressions introduced by our dependencies in the future, and we have to balance the risk of trusting dependencies to do their job correctly against the benefit of all the code they provide us. "Hell is other programmers", as Sartre would remind us.